crates/ark/src/r_task.rs

```diff
@@ -252,8 +254,7 @@ where
     F: FnOnce() -> Fut + 'static + Send,
     Fut: Future<Output = ()> + 'static,
 {
-    // Idle tasks are always run from the read-console loop
-    if unsafe { R_TASK_BYPASS } {
+    if harp::IS_TESTING {
         // Escape hatch for unit tests
         futures::executor::block_on(fun());
```

Comment:
This almost certainly needs the `R_TEST_LOCK` and `r_test_init()` too right? Even though we aren't using it yet.

Reply:
I left that part as is because it confused me and I did not get to the bottom of it, but it's currently only running on the main thread.

You're right we should be more careful though, because that might not be the case in the future, so we now take the lock here too.